pubtools/_pulp/tasks/set_maintenance/set_maintenance_off.py

```diff
@@ -0,0 +1,48 @@
+import logging
+import re
+
+from .base import SetMaintenance
+
+
+LOG = logging.getLogger("set-maintenance")
+
+
+class SetMaintenanceOff(SetMaintenance):
+    """Unset repositories maintenance mode.
+
+    See "pub maintenance-on --help" for more information on maintenance mode.
+    """
+
+    def run(self):
+        report = self.pulp_client.get_maintenance_report().result()
+
+        to_remove = []
+        if self.args.repo_regex:
+            for entry in report.entries:
+                if re.match(self.args.repo_regex, entry.repo_id):
+                    to_remove.append(entry.repo_id)
+        if self.args.repo_ids:
+            existed_repo_ids = [entry.repo_id for entry in report.entries]
+            for repo_id in self.args.repo_ids:
+
+                if repo_id in existed_repo_ids:
+                    to_remove.append(repo_id)
+                else:
+                    LOG.warn("Repository %s is not in Maintenance Mode", repo_id)
+
+        LOG.info(
+            "Following repositories will be removed from Maintenance Mode: \n%s",
+            "\n".join(to_remove),
+        )
+
+        report = report.remove(to_remove, owner=self.args.owner)
+
+        self.pulp_client.set_maintenance(report).result()
```

Comment:
Yup, it returns some tasks. I tried locally and seems like task ids are logged by pubtools-pulplib already, do you think we should log them again here?

Reply:
I think it's important to have some hint in the logs about what a task did, otherwise it will look like a bunch of meaningless indistinguishable IDs.  But I agree it doesn't look good to have all the tasks logged twice in different ways.

A couple of ideas what to do here:

1. Do what I suggested here (and always do that in pubtools-pulp), and downgrade the related log in pulplib to DEBUG only.
2. Or, don't add any logging here.  Instead we can enhance the pulplib logging to also include the task tags.  Tags typically contain enough info to distinguish different tasks (e.g. the tags will include "publish" and the repo ID, when publishing a repo).

Maybe you could come up with some other ideas too if you play around with it.  At the moment I'm thinking option (2) ought to work out well.  If you think so too then I'd withdraw this suggestion and we don't need to log task IDs here.

Reply:
yeah, I prefer to go with option 2